src/ApiService/ApiService/onefuzzlib/ImageOperations.cs

```diff
@@ -26,17 +27,22 @@ public interface IImageOperations {
 public class ImageOperations : IImageOperations {
     private IOnefuzzContext _context;
     private ILogTracer _logTracer;
+    private readonly IMemoryCache _cache;
 
     const string SubscriptionsStr = "/subscriptions/";
     const string ProvidersStr = "/providers/";
 
 
-    public ImageOperations(ILogTracer logTracer, IOnefuzzContext context) {
+    public ImageOperations(ILogTracer logTracer, IOnefuzzContext context, IMemoryCache cache) {
         _logTracer = logTracer;
         _context = context;
+        _cache = cache;
     }
 
-    public async Task<OneFuzzResult<Os>> GetOs(Region region, string image) {
+    public Task<OneFuzzResult<Os>> GetOs(Region region, string image)
+        => _cache.GetOrCreateAsync<OneFuzzResult<Os>>($"GetOs-{region}-{image}", entry => GetOsInternal(region, image));
```

Comment:
I've been using key classes for this, which helps to guarantee that we don't accidentally have a collision, e.g. `record class GetOsKey(Region Region, string Image);`